software/base/src/main/java/org/apache/brooklyn/entity/software/base/AbstractSoftwareProcessWinRmDriver.java

```diff
@@ -321,12 +332,21 @@ public int copyTo(InputStream source, String destination) {
     }
 
     public void rebootAndWait() {
+        rebootAndWait(null);
+    }
+
+    public void rebootAndWait(String hostname) {
         try {
-            executePsScriptNoRetry(ImmutableList.of("Restart-Computer -Force"));
+            if (hostname != null) {
+                getLocation().executePsScript(ImmutableMap.of(WinRmTool.COMPUTER_NAME, hostname), ImmutableList.of("Restart-Computer -Force"));
+            } else {
+                getLocation().executePsScript(ImmutableList.of("Restart-Computer -Force"));
+            }
         } catch (Exception e) {
-            // Restarting the computer will cause the command to fail; ignore the exception and continue
+            LOG.warn("Exception when restarting machine: ", e);
```

Comment:
My understanding was that this exception was now *not* expected, @bostko is that the case?